module/state_synchronization/indexer/persister_test.go

```diff
@@ -0,0 +1,278 @@
+	storagemock "github.com/onflow/flow-go/storage/mock"
+	"github.com/onflow/flow-go/storage/store/inmemory/unsynchronized"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+type PersisterSuite struct {
+	suite.Suite
+	persister              *Persister
+	inMemoryRegisters      *unsynchronized.Registers
+	inMemoryEvents         *unsynchronized.Events
+	inMemoryCollections    *unsynchronized.Collections
+	inMemoryTransactions   *unsynchronized.Transactions
+	inMemoryResults        *unsynchronized.LightTransactionResults
+	inMemoryTxResultErrMsg *unsynchronized.TransactionResultErrorMessages
+	registers              *storagemock.RegisterIndex
+	events                 *storagemock.Events
+	collections            *storagemock.Collections
+	transactions           *storagemock.Transactions
+	results                *storagemock.LightTransactionResults
+	txResultErrMsg         *storagemock.TransactionResultErrorMessages
+	batch                  *storagemock.Batch
+	database               *storagemock.DB
+	executionResult        *flow.ExecutionResult
+	header                 *flow.Header
+}
+
+func TestPersisterSuite(t *testing.T) {
+	t.Parallel()
+	suite.Run(t, new(PersisterSuite))
+}
+
+func (p *PersisterSuite) SetupTest() {
+	t := p.T()
+
+	p.header = unittest.BlockHeaderFixture()
+	p.inMemoryRegisters = unsynchronized.NewRegisters(p.header.Height)
+	p.inMemoryEvents = unsynchronized.NewEvents()
+	p.inMemoryCollections = unsynchronized.NewCollections()
+	p.inMemoryTransactions = unsynchronized.NewTransactions()
+	p.inMemoryResults = unsynchronized.NewLightTransactionResults()
+	p.inMemoryTxResultErrMsg = unsynchronized.NewTransactionResultErrorMessages()
+
+	p.registers = storagemock.NewRegisterIndex(t)
+	p.events = storagemock.NewEvents(t)
+	p.collections = storagemock.NewCollections(t)
+	p.transactions = storagemock.NewTransactions(t)
+	p.results = storagemock.NewLightTransactionResults(t)
+	p.txResultErrMsg = storagemock.NewTransactionResultErrorMessages(t)
+
+	p.batch = storagemock.NewBatch(t)
+	p.batch.On("Commit").Return(nil).Maybe()
+	p.batch.On("Close").Return(nil)
+	p.database = storagemock.NewDB(t)
+	p.database.On("NewBatch").Return(p.batch)
+
+	block := unittest.BlockWithParentFixture(p.header)
+	p.executionResult = unittest.ExecutionResultFixture(unittest.WithBlock(block))
+
+	p.persister = NewPersister(
+		zerolog.Nop(),
+		p.inMemoryRegisters,
+		p.inMemoryEvents,
+		p.inMemoryCollections,
+		p.inMemoryTransactions,
+		p.inMemoryResults,
+		p.inMemoryTxResultErrMsg,
+		p.registers,
+		p.events,
+		p.collections,
+		p.transactions,
+		p.results,
+		p.txResultErrMsg,
+		p.database,
+		p.executionResult,
+		p.header,
+	)
+
+	p.populateInMemoryStorages()
+}
+
+func (p *PersisterSuite) populateInMemoryStorages() {
+	regEntries := make(flow.RegisterEntries, 3)
+	for i := 0; i < 3; i++ {
+		regEntries[i] = unittest.RegisterEntryFixture()
+	}
+	p.Require().NoError(p.inMemoryRegisters.Store(regEntries, p.header.Height))
```

Comment:
nit: be consistent with error handling
```suggestion
	err := p.inMemoryRegisters.Store(regEntries, p.header.Height)
	p.Require().NoError(err)
```